packetbeat/flows/worker.go

```diff
@@ -272,22 +330,43 @@ func (fw *flowsProcessor) execute(w *worker, checkTimeout, handleReports, lastRe
 
 			if reportFlow {
 				debugf("report flow")
-				fw.report(w, ts, flow, isOver, intNames, uintNames, floatNames)
+				fw.report(w, ts, flow, isOver, intNames, uintNames, floatNames, killReason)
 			}
 		}
 	}
 
 	fw.spool.flush()
 }
 
-func (fw *flowsProcessor) report(w *worker, ts time.Time, flow *biFlow, isOver bool, intNames, uintNames, floatNames []string) {
-	event := createEvent(fw.watcher, ts, flow, isOver, intNames, uintNames, floatNames)
+func shouldEndFlow(flow *biFlow, fw *flowsProcessor, ts time.Time, activeFlowTimeout bool) (flowEndReason, bool) {
```

Comment:
Im a little confused here - what if there are two different timeouts; one for inactive and one for active. how do we account for those two different values?

Reply:
That is handled at a different place .That is handled in makeWorker function based on `timeout, period, activeTimeout`  . However these is a bug I need to change the if condition 
```
if checkTimeout {
```
to 
```
if checkTimeout || handleActiveTimeout {
```

Reply:
I ended up actually checking the check altogether